core/js/src/main/scala/tf/bug/dirs/os.scala.scala

```diff
@@ -0,0 +1,13 @@
+package tf.bug.dirs
+
+import scala.scalajs.js
+import scala.scalajs.js.annotation.JSImport
+
+/**
+  * Necessary facades around the Node.js `os` module for discovering dirs
+  */
+object os {
```

Comment:
I wonder if this can be made an internal object to `MacPlatform`?

Reply:
For an eventual Linux platform, it would be nice to have access to that function, no?

Reply:
I guess so.

It's weird that `user.home` doesn't work on these platforms...

Also, I just noticed this file has `.scala.scala` at the end of it.

Reply:
I should stop programming at night! Thank you for spotting that haha.

> It's weird that user.home doesn't work on these platforms...
What do you mean with this?

Reply:
I would have assumed the same `getProperty("user.home")` would've worked on all of JVM/JS/Native, especially because I used `os.name` elsewhere... although I did never verify that it actually worked....

Reply:
I don't think that is the responsibility of Scala.js per se, it just compiles Scala to JS. Bindings to Node api's seem out of scope.
I tried the following with scala-cli:

```scala
//> using scala "3.2.1"
@main def printHome(): Unit =
  println(System.getProperty("user.home"))
```
and running `scala-cli Main.scala --js` prints `null`. `fs2-io` also wraps the Node.js function.
On top of that in the browser this API would fail.

Reply:
Nobody should be using this in the browser regardless...

`user.home` and `os.name` not being filled in is a little irking. I do think we'll need a platform-specific object for those then, independent of OS. JVM can just call `getProperties`, JS can use the Node bindings available... Native is still a tricky one.

Reply:
Yea I understand the irkyness of it.

> a platform-specific object for those then

Like a `trait Target` that contains something like `def homeDir` and `def osName`?

Reply:
Exactly 👍 

And honestly, probably better to provide `Platform` over `osName: String`.